Fixes some broken clothing equip delays

Tamamitsune opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

~~Updates the equip delays for all items to be slightly faster, and~~ allows equipping armor while moving (not hardsuits).
Also, fixes a number of factional suit items using the wrong equip delay values

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

~~Some of the equip delays were pretty dang long and felt honestly just clunky. While you don't put on/off clothes too often, it did feel silly when just slipping on gloves took over a second, or putting on/removing a jacket took two whole seconds.~~
A decent number of factional suits were using the hard-suit equip delay and restrictions, when half the time you were just tossing on a jacket.

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog

:cl:
fix: Fixed some factional suit items using the wrong equip delay
balance: You can now move while equipping armor, such as bulletproof vests.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Tamamitsune committed:
Decreases most clothing equip delays

Tamamitsune committed:
Merge branch 'shiptest-ss13:master' into clothingequipbuff

Tamamitsune committed:
Merge branch 'shiptest-ss13:master' into clothingequipbuff

thgvr commented:
Afraid this isn't the answer. If there is an item that feels like it's taking too long to equip, there is a really good chance it is using the wrong delay. This stuff was carefully tuned and intended so you can't just swap your gear on a whim and give some importance to the act of gearing up before an expedition from your ship.

Tamamitsune committed:
Reverts defines edits, adds more appropriate equip_delay flags to factional suits

Tamamitsune commented:
> Afraid this isn't the answer. If there is an item that feels like it's taking too long to equip, there is a really good chance it is using the wrong delay. This stuff was carefully tuned and intended so you can't just swap your gear on a whim and give some importance to the act of gearing up before an expedition from your ship.

Fair enough, I went a little overboard on a number of the tweaks I'll admit. I reverted those defines changes, and went through and found a number of items using the wrong delay and fixed those up.

FalloutFalcon commented on code/modules/clothing/factions/hardliners.dm:
```diff
@@ -111,6 +118,13 @@
 	armor = list("melee" = 35, "bullet" = 35, "laser" = 35, "energy" = 40, "bomb" = 25, "bio" = 0, "rad" = 0, "fire" = 50, "acid" = 50, "wound" = 10)
 	togglename = "buttons"
 
+	equipping_sound = EQUIP_SOUND_MED_GENERIC
+	unequipping_sound = UNEQUIP_SOUND_MED_GENERIC
+	equip_delay_self = EQUIP_DELAY_SUIT
+	equip_delay_other = EQUIP_DELAY_SUIT * 1.5
+	strip_delay = EQUIP_DELAY_SUIT * 1.5
+	equip_self_flags = EQUIP_ALLOW_MOVEMENT | EQUIP_SLOWDOWN
+
 /obj/item/clothing/suit/toggle/armor/vest/hardliners/Initialize()
```
I know this is a jacket but I would assume anything with decent armor should have the balance of armor?

Tamamitsune replied:
Sorry for hella late reply, had two rough work weeks in a row; but yeah i agree, those lines make it so it does have the typical armor delay on it, there wasnt a base clothing/suit/toggle/armor to inherit from, just the clothing/suit/toggle giving it the standard jacket one. Would it be better to add clothing/suit/toggle/armor to inherit from?

FalloutFalcon commented on code/modules/clothing/suits/armor.dm:
```diff
@@ -22,7 +22,7 @@
 	equip_delay_other = EQUIP_DELAY_SUIT * 1.5
 	strip_delay = EQUIP_DELAY_SUIT * 1.5
-	equip_self_flags = null
+	equip_self_flags = EQUIP_ALLOW_MOVEMENT | EQUIP_SLOWDOWN
```
Will need to get thgvr's opinion of whether this can actually be left in.

Tamamitsune replied:
fair enough, i personally just feel its a bit odd/stiff being forced to stand still for that, when i could feasibly see most of the armor items that aren't full suits not completely inhibiting movement while trying to fit 'em, and i feel the slowdown and time it takes already to actually fully put them on stops any issues concerning balance

thgvr commented:
we'll see how it goes
